client/client.go

```diff
@@ -0,0 +1,425 @@
+	if err != nil {
+		return fmt.Errorf("get engine id by name: %w", err)
+	}
+
+	c.engineEndpoint, err = c.getEngineURLByName(ctx)
+	if err != nil {
+		return fmt.Errorf("get engine url by name: %w", err)
+	}
+
+	return nil
+}
+
+// StartEngine starts a Firebolt engine and returns
+// a bool indicating whether the engine is started or not.
+func (c *Client) StartEngine(ctx context.Context) (bool, error) {
+	if c.accountID == "" || c.engineID == "" {
+		return false, errAccountIDOrEngineIDIsEmpty
+	}
+
+	req, err := c.newRequest(ctx, http.MethodPost, fmt.Sprintf(startEngineURL, c.accountID, c.engineID), nil)
+	if err != nil {
+		return false, fmt.Errorf("create start engine request: %w", err)
+	}
+
+	var engineResponse engineResponse
+	_, err = c.do(ctx, req, &engineResponse)
+	if err != nil {
+		return false, fmt.Errorf("do start engine request: %w", err)
+	}
+
+	isEngineStarted := engineResponse.Engine.CurrentStatus == EngineStartedStatus
+
+	return isEngineStarted, nil
+}
+
+// RunQuery runs an SQL query.
+func (c *Client) RunQuery(ctx context.Context, query string) (*RunQueryResponse, error) {
+	b := bytes.NewBuffer([]byte(query))
+
+	req, err := c.newRequest(ctx, http.MethodPost, fmt.Sprintf("https://%s/?database=%s",
+		c.engineEndpoint, c.dbName), b)
+	if err != nil {
+		return nil, fmt.Errorf("create run query request: %w", err)
+	}
+
+	var resp RunQueryResponse
+	_, err = c.do(ctx, req, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("do run query request: %w", err)
+	}
+
+	return &resp, nil
+}
+
+// GetEngineStatus returns the current status of the underlying engine.
+func (c *Client) GetEngineStatus(ctx context.Context) (string, error) {
+	if c.accountID == "" || c.engineID == "" {
+		return "", errAccountIDOrEngineIDIsEmpty
+	}
+
+	engineResponse, err := c.getEngineByID(ctx)
+	if err != nil {
+		return "", fmt.Errorf("get engine by id: %w", err)
+	}
+
+	return engineResponse.Engine.CurrentStatus, nil
+}
+
+// RefreshToken performs a refresh token request.
+// The method set the *Client.accessToken field to the new access token.
+func (c *Client) RefreshToken(ctx context.Context) error {
+	refreshTokenRequest := refreshTokenRequest{
+		RefreshToken: c.refreshToken,
+	}
+
+	req, err := c.newRequest(ctx, http.MethodPost, refreshTokenURL, &refreshTokenRequest)
+	if err != nil {
+		return fmt.Errorf("create refresh token request: %w", err)
+	}
+
+	var loginResp loginResponse
+	_, err = c.do(ctx, req, &loginResp)
+	if err != nil {
+		return fmt.Errorf("do refresh token request: %w", err)
+	}
+
+	c.accessToken = loginResp.AccessToken
+
+	return nil
+}
+
+// WaitEngineStarted periodically checks the engine status,
+// and if the status is equal to ENGINE_STATUS_RUNNING_REVISION_SERVING or ctx is canceled returns.
+// It's a blocking method.
+func (c *Client) WaitEngineStarted(ctx context.Context) error {
+	ticker := time.NewTicker(engineStatusCheckTimeout)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
```

Comment:
Returning nil here, would indicate to the caller of this method, that the engine is started, no? I'd suggest returning ctx.Error() here.

Reply:
done